src/glue/jobs/json_to_parquet.py

```diff
@@ -0,0 +1,319 @@
+    "healthkitv2workouts": ["HealthKitWorkoutKey"],
+    "healthkitv2activitysummaries": ["HealthKitActivitySummaryKey"],
+    "googlefitsamples": ["GoogleFitSampleKey"],
+    "symptomlog": ["DataPointKey"]
+}
+
+def get_args() -> tuple[dict, dict]:
+    """
+    Get job and workflow arguments.
+
+    Returns:
+        (tuple) : (job arguments (dict), workflow run properties (dict))
+    """
+    glue_client = boto3.client("glue")
+    args = getResolvedOptions(
+             sys.argv,
+             ["WORKFLOW_NAME",
+              "WORKFLOW_RUN_ID",
+              "JOB_NAME",
+              "glue-table"])
+    workflow_run_properties = glue_client.get_workflow_run_properties(
+            Name=args["WORKFLOW_NAME"],
+            RunId=args["WORKFLOW_RUN_ID"])["RunProperties"]
+    return args, workflow_run_properties
+
+def has_nested_fields(schema: StructType) -> bool:
+    """
+    Determine whether a DynamicFrame schema has struct or array fields.
+
+    If the DynamicFrame does not have fields of these data types, it is flat and
+    can be written directly to S3.  If it does, then the DynamicFrame will need
+    to be 'relationalized' so that all the data contained in the DynamicFrame has
+    been flattened.
+
+    Args:
+        schema (awsglue.gluetypes.StructType): The schema of a DynamicFrame.
+
+    Returns:
+        bool: Whether this schema contains struct or array fields.
+    """
+    for col in schema:
+        if col.dataType.typeName() in ["array", "struct"]:
+            return True
+    return False
+
+def get_table(
+        table_name: str,
+        database_name: str,
+        glue_context: GlueContext
+    ) -> DynamicFrame:
+    """
+    Return a table as a DynamicFrame with an unambiguous schema.
+
+    Args:
+        table_name (str): The name of the Glue table.
+        database_name (str): The name of the Glue database
+        glue_context (GlueContext): The glue context
+
+    Returns:
+        awsglue.DynamicFrame: The Glue Table's data as a DynamicFrame
+            after the duplicates have been dropped by descending
+            export date.
+    """
+    _, table_data_type = table_name.split("_")
+    table = (
+            glue_context.create_dynamic_frame.from_catalog(
+                 database=database_name,
+                 table_name=table_name,
+                 transformation_ctx="create_dynamic_frame"
+            )
+            .resolveChoice(
+                choice="match_catalog",
+                database=database_name,
+                table_name=table_name
+            )
+    )
+    spark_df = table.toDF()
+    table = DynamicFrame.fromDF(
+            dataframe=(
+                spark_df
+                .sort(spark_df.export_end_date.desc())
+                .dropDuplicates(
+                    subset=INDEX_FIELD_MAP[table_data_type]
+                )
+            ),
+            glue_ctx=glue_context,
+            name=table_name
+    )
+    return table
+
+def write_table_to_s3(
```

Comment:
Are some of these functions used across BridgeDownstream and RECOVER?  I wonder if it'd be worth creating a package.  Can you install custom packages in AWS Glue?

Reply:
> Are some of these functions used across BridgeDownstream and RECOVER? I wonder if it'd be worth creating a package.
Yes, we will want to maintain a separate python library once we have a more generalized framework for data to parquet. I'm not sure if it makes sense doing this for RECOVER considering the tight deadline we are on, but it's worth keeping in mind for future projects that require a data to compute of some sort.